client/components/Editor/index.jsx

```diff
@@ -48,10 +61,39 @@ class Editor extends Component {
 
     onClickScenarioAction(event, data) {
         if (data.name === 'save-scenario') {
-            // TODO: Determine if we need to actually
-            //       save the scenario, or if the
-            //       auto-save is sufficient.
-            this.updateEditorMessage('Teacher Moment saved');
+            this.updateScenario();
+        }
+
+        if (data.name === 'save-status') {
+            const scenario = Object.assign(
+                {},
+                {
+                    ...this.state.scenario,
+                    status: data.id
+                }
+            );
+
+            this.setState({ scenario }, () => {
+                this.updateScenario(scenario);
+            });
+        }
+    }
+
+    async fetchScenario() {
+        const response = await (await fetch(
+            `/api/scenarios/${this.state.scenarioId}`
+        )).json();
+
+        const { scenario } = response;
+        if (response.status === 200) {
+            this.props.setScenario({
+                title: scenario.title,
+                description: scenario.description,
+                categories: scenario.categories,
+                status: scenario.status
+            });
+
+            this.setState({ scenario });
```

Comment:
An example of the two state systems (`setScenario` and `setState`), though it's good they're being kept in sync here. Just not sure we need both.